3-extra/1-factorial.js

```diff
@@ -9,7 +9,11 @@
 */
 
 function factorial(input) {
-    // TODO
+    
+    let factor = 1;
+    for (let i = input; i > 0; i--)
+        factor *= i;
+    return factor;
```

Comment:
looks good - how could you do this without mutating a variable? (e.g. without doing `i--` or `factor *=`) https://www.freecodecamp.org/news/how-to-factorialize-a-number-in-javascript-9263c89a4b38/